library/core/benches/num/int_sqrt/mod.rs

```diff
@@ -0,0 +1,207 @@
+//! This benchmarks the `Integer::isqrt` methods.
```

Comment:
It looks like the macros are more or less copied between `benches` and `benches_nonzero`. Could this be deduplicated? Same for tests, the comments are repeated four places so it seems easy to get out of sync.

One pattern to help with things like this is for the macro to take a `$ctor:expr` metavar, which would just be `|v| v` for integers and `|v| NonZero::new(v).unwrap()` for nonzero types. Or return `Some(v)` and don't `unwrap()` so you can filter invalid values while building the benches.

Reply:
Related, if you can pull some of the macro logic out to discrete functions then it is usually easier to read. You might be able to have a `fn make_test_values(max_sqrt: u128, bits: u32, max_leading_zeros: u32, ...) -> Vec<u128>`, then within the macro you only have to convert from `Vec<u128>` to `Vec<T>`.

(Low priority.)

Reply:
OK, I'll do that.

Reply:
OK, I've replaced my implementation with the tests for `ilog10`, as they have three separate benchmarks per type.

Reply:
I also added a test that uses a uniform distribution rather than the exponential distribution in the existing tests.